SocialCareCaseViewerApi.Tests/V1/UseCase/AllocationsUseCaseTests.cs

```diff
@@ -78,12 +78,97 @@ public void ListAllocationsByWorkerEmail()
+                .Returns(gatewayResponse);
+
+            var response = _allocationsUseCase.Execute(request);
+
+            response.Allocations.Should().BeEquivalentTo(gatewayResponse);
+        }
+
+        [Test]
+        public void ListAllocationsByTeamIdReturnsNothingIfNoMatches()
+        {
+            var request = new ListAllocationsRequest() { TeamId = 8 };
+            var gatewayResponse = new List<Allocation> { new Allocation() { Id = 1, PersonId = 2, AllocatedWorker = "Test Worker" } };
+
+            _mockDatabaseGateway.Setup(x => x.SelectAllocations(0, 0, null, 10))
+                .Returns(gatewayResponse);
+
+            var response = _allocationsUseCase.Execute(request);
+
+            response.Allocations.Should().BeNullOrEmpty();
+        }
+
+        [Test]
+        public void ListAllocationsByMosaicIdReturnsExpectedList()
+        {
+            var request = new ListAllocationsRequest() { MosaicId = 3 };
+            var gatewayResponse = new List<Allocation> { new Allocation() { Id = 1, PersonId = 2, AllocatedWorker = "Test Worker" } };
+
+            _mockDatabaseGateway.Setup(x => x.SelectAllocations(3, 0, null, 0))
+                .Returns(gatewayResponse);
+
+            var response = _allocationsUseCase.Execute(request);
+
+            response.Allocations.Should().BeEquivalentTo(gatewayResponse);
+        }
+
+        [Test]
+        public void ListAllocationsByMosaicIdReturnsNothingIfNoMatches()
+        {
+            var request = new ListAllocationsRequest() { MosaicId = 3 };
+            var gatewayResponse = new List<Allocation> { new Allocation() { Id = 1, PersonId = 2, AllocatedWorker = "Test Worker" } };
+
+            _mockDatabaseGateway.Setup(x => x.SelectAllocations(3, 0, null, 3))
+                .Returns(gatewayResponse);
+
+            var response = _allocationsUseCase.Execute(request);
+
+            response.Allocations.Should().BeNullOrEmpty();
+        }
+        [Test]
+        public void ListAllocationsByWorkerIdReturnsExpectedList()
+        {
+            var request = new ListAllocationsRequest() { WorkerId = 5 };
+            var gatewayResponse = new List<Allocation> { new Allocation() { Id = 1, PersonId = 2, AllocatedWorker = "Test Worker" } };
+
+            _mockDatabaseGateway.Setup(x => x.SelectAllocations(0, 5, null, 0))
+                .Returns(gatewayResponse);
+
+            var response = _allocationsUseCase.Execute(request);
+
+            response.Allocations.Should().BeEquivalentTo(gatewayResponse);
+        }
+
+        [Test]
+        public void ListAllocationsByWorkerIdReturnsNothingIfNoMatches()
+        {
+            var request = new ListAllocationsRequest() { MosaicId = 5 };
+            var gatewayResponse = new List<Allocation> { new Allocation() { Id = 1, PersonId = 2, AllocatedWorker = "Test Worker" } };
+
+            _mockDatabaseGateway.Setup(x => x.SelectAllocations(0, 5, null, 3))
```

Comment:
As above.

Reply:
Done on [d50eb72](https://github.com/LBHackney-IT/social-care-case-viewer-api/pull/610/commits/d50eb727ec51d87e296be9b85f1392ebfb7118f4)